src/Misc/layoutbin/powershell/Add-XamarinAndroidCapabilities.ps1

```diff
@@ -1,4 +1,19 @@
 [CmdletBinding()]
 param()
 
-$null = Add-CapabilityFromRegistry -Name 'Xamarin.Android' -Hive 'LocalMachine' -View 'Registry32' -KeyName 'Software\Novell\Mono for Android' -ValueName 'InstalledVersion'
+if (!(Add-CapabilityFromRegistry -Name 'Xamarin.Android' -Hive 'LocalMachine' -View 'Registry32' -KeyName 'Software\Novell\Mono for Android' -ValueName 'InstalledVersion')) {
+    $vs15 = Get-VisualStudio_15_0
+    if ($vs15 -and $vs15.installationPath) {
+        # End with "\" for consistency with old ShellFolder values.
+        $shellFolder15 = $vs15.installationPath.TrimEnd('\'[0]) + "\"
+        $xamarinAndroidDir = ([System.IO.Path]::Combine($shellFolder15, 'MSBuild', 'Xamarin', 'Android')) + '\'
+        if ((Test-Container -LiteralPath $xamarinAndroidDir)) {
+            $versionFile = ([System.IO.Path]::Combine($xamarinAndroidDir, 'Version'))
+            $version = Get-Content $versionFile
+            if ($version) {
+                Write-Capability -Name 'Xamarin.Android' -Value $version
+            }
+        }
+    }
+}
+
```

Comment:
Right, we don't use this version number at all.  I am favoring existing proven code over the new method.